documentation/wiki/Results-Cache.md

```diff
@@ -0,0 +1,58 @@
+
+![MSBuild Cache Flow](CacheFlow.png)
+
+## `ResultsCache` (The Core Cache Component)
+
+`ResultsCache` is the primary storage mechanism where MSBuild keeps the outcomes of its build targets.
+
+### What It Does
+
+1.  **Stores and Retrieves Build Outcomes**:
+    *   It holds onto `BuildResult` objects. Each `BuildResult` is a record of the outcome for one or more build steps (targets) for a specific project setup (configuration).
+    *   These outcomes are organized using a special ID (`configurationId`) that represents the unique way a project was defined (like list of targets, tools versions, etc).
+    *   When MSBuild is about to build something, it first queries this cache: "Have I processed this exact task before with these exact settings?" If the cache has a valid outcome, MSBuild can use it.
+    *   When new build work is completed, the outcomes are added to this cache so they can be used in the future. If outcomes for that specific setup already exist, the new information is merged with the old.
+
+2.  **Checks if a Build Request Can Be Fulfilled from Cache**:
+    *   Before actually starting a build task, MSBuild uses this cache to see if all the necessary pieces have already been successfully built and their stored outcomes are still valid.
+    *   This check considers the specific build settings and whether detailed project information (like all its properties and items) is needed from the cached outcome. If everything checks out, the build step can be satisfied from the cache.
+
+3.  **Clears Out Stored Outcomes**:
+    *   Sometimes, the cache needs to be cleared. `ResultsCache` can remove all the outcomes it has stored.
+    *   It can also clear out outcomes for just one specific project setup, for example, if project files have changed and the old outcomes are no longer valid.
+
+4.  **Prepares Cache Data for Transfer**:
+    *   `ResultsCache` can convert the information it stores (either all of it or specific `BuildResult` records) into a special format.
+    *   This format can then be sent to other MSBuild processes.
+
+### How `BuildRequestDataFlags` Affect Caching
+
+When MSBuild decides to build something, the request (`BuildRequest`) carries special instructions called `BuildRequestDataFlags`. These flags tell MSBuild *how* to perform the build and *what kind of result* is expected. The `ResultsCache` pays close attention to these flags because they are crucial for determining if a cached outcome is truly a match for a new request.
+It's an out-in feature that can be disabled with change wave `17.12`. See more details in [ChangeWaves.md](ChangeWaves.md)
+
+Here's how some important flags interact with caching:
+
+*   **Flags Affecting Build Behavior:** - if any of the flags is presented only in request or cache - cache miss is returned:
```

Comment:
nit: the formatting `-...-` looks a little off to me here `- if any of the flags is presented only in request or cache -`. as a suggestion, use brackets instead?